tests/unit/commands/test_set_default_track.py

```diff
@@ -43,17 +43,23 @@ def test_set_default_track_without_login_must_ask(self):
         ]
 
         result = self.run_command(
-            ["set-default-track", "snap-test", "track"],
+            ["set-default-track", "snap-test", "2.0"],
             input="user@example.com\nsecret\n",
         )
         self.assertThat(
             result.output, Contains("You are required to login before continuing.")
         )
 
     def test_set_default_track(self):
-        result = self.run_command(["set-default-track", "snap-test", "track"])
+        result = self.run_command(["set-default-track", "snap-test", "2.0"])
 
         self.assertThat(result.exit_code, Equals(0))
         self.fake_metadata.mock.assert_called_once_with(
-            snap_name="snap-test", metadata=dict(default_track="track"), force=True
+            snap_name="snap-test", metadata=dict(default_track="2.0"), force=True
         )
+
+    def test_invalid_track_fails(self):
+        result = self.run_command(["set-default-track", "snap-test", "3.0"])
+
+        self.assertThat(result.exit_code, Equals(2))
+        self.assertThat(result.output, Contains("'2.0', 'latest'"))
```

Comment:
This could be improved to verify all of `brief`, `resolution` and `details`.

Reply:
This is exactly what I do not want to test, if we ever change the message output in exit_error, the ripple effect from this would eventually just be too cumbersome.

Reply:
I was thinking something like:

```
    @mock.patch("snapcraft.cli.store.echo.exit_error")
    def test_invalid_track_fails(self, mock_exit):
        self.run_command(["set-default-track", "snap-test", "3.0"])

        assert mock_exit.mock_calls == [
            mock.call(
                brief="The specified track does not exist for 'snap-test'.",
                details="Valid tracks: '2.0', 'latest'.",
                resolution="Ensure the track exists and try again.",
            )
        ]
```

Reply:
That verifies all of the strings under test, not what exit_error does with them.

Reply:
I will think about using a spy and not this exact method as changing this cripples click's test runner